### ISSUE TITLE
addToDictionary command not working

### ISSUE THREAD

**Issue by patrickswijgman** [vitallium/zed-ltex#4]
Hey! Maybe I'm missing something but it looks like the `Add to dictionary` command does not work.

I see this in the RPC messages when looking in the logs:

```json
// Send:
{"jsonrpc":"2.0","id":386,"method":"workspace/executeCommand","params":{"command":"_ltex.addToDictionary","arguments":[{"uri":"file://__REDACTED__.md","words":{"en-US":["__REDACTED__"]}}]}}
// Receive:
{"jsonrpc":"2.0","id":386,"result":{"success":false,"errorMessage":"Unknown command \u0027_ltex.addToDictionary\u0027, ignoring"}}
```

I guess I could add my custom words in the Zed config, but this code action was given so I would've hoped it worked :innocent: 

I don't know if this is an issue within Zed extension or an issue within `ltex-ls` itself. If you need some more info let me know :smile:

**Comment by vitallium** [comment]
Yeah, I also encountered that. I think this an issue with `ltex-plus` itself since it provides code actions as commands but [it does not handle them](https://github.com/ltex-plus/ltex-ls-plus/blob/develop/src/main/kotlin/org/bsplines/ltexls/server/LtexWorkspaceService.kt#L70):

```
{"jsonrpc":"2.0","id":13,"method":"workspace/executeCommand","params":{"command":"_ltex.addToDictionary","arguments":[{"uri":"file:///Users/vslobodin/Development/ruby-lsp/README.md","words":{"en-US":["userrs"]}}]}}
// Receive:
{"jsonrpc":"2.0","id":13,"result":{"success":false,"errorMessage":"Unknown command \u0027_ltex.addToDictionary\u0027, ignoring"}}
```

**Comment by 497e0bdf29873** [comment]
There seems to be something more wrong than ltex-plus just not inserting entries into the dictionary: it doesn't even load them. Due to this issue, I tried to pass it my old aspell database as instructed in the documentation (`":path"` as a word in the settings), but, no, it doesn't load it. It does understand individual words passed in the settings, though, so I think I got the setting correct.

**Comment by vitallium** [comment]
@497e0bdf29873 do you mind sharing your settings, please, so that I can take a look?

**Comment by 497e0bdf29873** [comment]
All settings or just `ltex`? The LTeX settings are

```
  "lsp": {
    "ltex": {
      "settings": {
        "ltex": {
          "language": "en-GB",
          "disabledRules": { "en-GB": ["OXFORD_SPELLING_Z_NOT_S"] },
          "dictionary": { "en-GB": ["subdifferential", ":~/lib/ltex-en-GB.txt"] }
        }
      }
    }
  }
```

The `lib-en-GB.txt` contains just the line `subdifferentials`.
The `subdifferential` in the `settings.json` is correctly recognised, but `subdifferentials` from the file not.

**Comment by tedzards509** [comment]
Typst runs the language servers in their own sandbox. Probably its possible to place the dictionary near the executable in the working directory of the server and somehow reference it there. More experimentation is needed though.

**Comment by weirane** [comment]
> All settings or just `ltex`? The LTeX settings are
> 
> ```
>   "lsp": {
>     "ltex": {
>       "settings": {
>         "ltex": {
>           "language": "en-GB",
>           "disabledRules": { "en-GB": ["OXFORD_SPELLING_Z_NOT_S"] },
>           "dictionary": { "en-GB": ["subdifferential", ":~/lib/ltex-en-GB.txt"] }
>         }
>       }
>     }
>   }
> ```
> 
> The `lib-en-GB.txt` contains just the line `subdifferentials`. The `subdifferential` in the `settings.json` is correctly recognised, but `subdifferentials` from the file not.

ltex-plus doesn't load external dictionaries, and other editor plugins have implemented their own mechanism of supporting external dictionaries: https://github.com/ltex-plus/ltex-ls-plus/issues/56. So this may also be an upstream issue.

I tried adding similar support to this Zed extension, but it seems like zed restricts file reads outside the worktree, so we can't read external files like `~/lib/ltex-en-GB.txt` directly. One workaround would be to declare the [process:exec](https://zed.dev/docs/extensions/capabilities#processexec) capability and use `/bin/cat` to read the file, but it feels a bit hacky.
